sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeLocalShuffleReader.scala

```diff
@@ -88,7 +88,7 @@ case class OptimizeLocalShuffleReader(conf: SQLConf) extends Rule[SparkPlan] {
     }
     val numReducers = shuffleDep.partitioner.numPartitions
     val expectedParallelism = advisoryParallelism.getOrElse(numReducers)
-    val numMappers = shuffleDep.rdd.getNumPartitions
+    val numMappers = math.max(1, shuffleDep.rdd.getNumPartitions)
```

Comment:
Hm, but if numMappers = 0, Array.fill never evaluates anything; you get an empty array. Are you sure this is the cause?

Reply:
yes, we may does not need this check. The error is tested with previous commit and it seems fixed in the latest commit. And I will close this PR. Thanks